src/components/drawer/drawer.component.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+
+import {
+  StyledDrawerWrapper,
+  StyledDrawerContent,
+  StyledIconButton,
+  StyledDrawerChildren,
+  StyledDrawerSidebar
+} from './drawer.style';
+
+const Drawer = ({
+  defaultIsExpanded,
+  expanded,
+  onChange,
+  children,
+  expandedSettings,
+  sidebar,
+  ...props
+}) => {
+  const isControlled = expanded !== undefined;
+  const [isExpandedInternal, setIsExpandedInternal] = useState(defaultIsExpanded || false);
+  const [firstOpen, setFirstOpen] = useState(true);
+  const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const toggleDrawer = useCallback((ev) => {
+    if (!isControlled) setIsExpandedInternal(!isExpanded);
+    if (onChange) onChange(ev, !isExpanded);
+    setFirstOpen(false);
+  }, [isControlled, isExpanded, onChange]);
+
+  const closedClass = firstOpen ? '' : 'closed';
+
+  return (
+    <StyledDrawerWrapper
+      isExpanded={ isExpanded }
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedSettings ? expandedSettings.maxWidth : '40%' }
+        className={ isExpanded ? 'open' : closedClass }
+      >
+        <StyledIconButton onAction={ toggleDrawer }>
+          <Icon type='chevron_right' />
+        </StyledIconButton>
+        <StyledDrawerSidebar>{ sidebar }</StyledDrawerSidebar>
+      </StyledDrawerContent>
+      <StyledDrawerChildren>
+        {children}
+      </StyledDrawerChildren>
+    </StyledDrawerWrapper>
+  );
+};
+
+Drawer.propTypes = {
+  children: PropTypes.node,
+  /** Set the default state of expansion of the Drawer if component is meant to be used as uncontrolled */
+  defaultIsExpanded: PropTypes.bool,
+  /** Sets the expansion state of the Drawer if component is meant to be used as controlled */
+  expanded: PropTypes.bool,
+  /** Callback fired when expansion state changes, onChange(event: object, isExpanded: boolean) */
+  onChange: PropTypes.func,
+  sidebar: PropTypes.node,
+  expandedSettings: PropTypes.object
```

Comment:
Replace the values with the real ones from DLS.
```suggestion
  expandedWidth: PropTypes.oneOf(['40%', '50%', '60%'])
```

Reply:
after chatting to Harpal, we decided to go % and px way., i set the propType to string.